.gitlab/e2e.yml

```diff
@@ -2,11 +2,18 @@
 # e2e stage
 # Contains jobs which runs e2e tests on our Docker images.
 
+.container_e2e_needs:
+  needs:
+    - qa_agent
+    - qa_dca
+    - qa_dogstatsd
+
 .k8s_e2e_template:
   stage: e2e
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/gitlab_agent_deploy:$DATADOG_AGENT_BUILDERS
   tags: ["arch:amd64"]
   dependencies: []
+  extends: .container_e2e_needs
```

Comment:
There is some confusion between the legacy and the new e2e tests here.

### Legacy e2e tests are using

* `datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py3`
* `datadog/cluster-agent-dev:${CI_COMMIT_REF_SLUG}`

images. See [line 37](https://github.com/DataDog/datadog-agent/pull/22817/files#diff-ef82ddf42fac4f3d2cea4532ca305be2ca01345a7f3a471ffb6652f8a5feded5R37) for ex.

Those images are published by the
* `dev_branch_multiarch-a7` https://github.com/DataDog/datadog-agent/blob/0be86ff70fe5748cac248b53f90b4cd8e5eba150/.gitlab/dev_container_deploy/docker_linux.yml#L71
* `dca_dev_branch` https://github.com/DataDog/datadog-agent/blob/0be86ff70fe5748cac248b53f90b4cd8e5eba150/.gitlab/dev_container_deploy/docker_linux.yml#L145
jobs

### New e2e tests are using

* `669783387624.dkr.ecr.us-east-1.amazonaws.com/agent:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}`
* `669783387624.dkr.ecr.us-east-1.amazonaws.com/cluster-agent:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}`
* `669783387624.dkr.ecr.us-east-1.amazonaws.com/dogstatsd:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}`

images.

Those images are indeed published by
* `qa_agent`
* `qa_dca`
* `qa_dogstatsd`
jobs.

## Suggestion:

```suggestion
.container_new_e2e_needs:
  needs:
    - qa_agent
    - qa_dca
    - qa_dogstatsd

.k8s_e2e_template:
  stage: e2e
  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/gitlab_agent_deploy:$DATADOG_AGENT_BUILDERS
  tags: ["arch:amd64"]
  dependencies: []
  needs:
    - dev_branch_multiarch-a7
    - dca_dev_branch
```

Reply:
Added the correct dependencies for the `legacy e2e`